### PR TITLE
Add Kumaraswamy and relaxed Bernoulli distributions

### PR DESCRIPTION
**Pull request by fehiepsi**
This PR adds Kumaraswamy and RelaxedBernoulli distributions. Those distributions are useful to approximate `Beta` and `Bernoulli` - they can be easily reparameterized.

For RelaxedBernoulli, I don't support `probs` argument to simplify the implementation.

### CODE REVIEW

File: numpyro/distributions/continuous.py [hunk 1]
```diff
@@ -505,6 +508,60 @@ def icdf(self, q):
         return self.loc - self.scale * jnp.log(-jnp.log(q))
 
 
+class Kumaraswamy(TransformedDistribution):
```
Comment (OlaRonning) [inline numpyro/distributions/continuous.py:511]: I checked math against wiki. Look good!

File: numpyro/distributions/continuous.py [hunk 2]
```diff
@@ -1426,6 +1483,28 @@ def tree_flatten(self):
         return super(TransformedDistribution, self).tree_flatten()
 
 
+class RelaxedBernoulliLogits(TransformedDistribution):
```
Comment (OlaRonning) [inline numpyro/distributions/continuous.py:1486]: I checked against `tfp` implementation. Looks good.

File: test/test_distributions.py [hunk 3]
```diff
@@ -285,6 +285,8 @@ def get_sp_dist(jax_dist):
     T(_ImproperWrapper, constraints.positive, (), (3,)),
     T(dist.InverseGamma, np.array([1.7]), np.array([[2.0], [3.0]])),
     T(dist.InverseGamma, np.array([0.5, 1.3]), np.array([[1.0], [3.0]])),
+    T(dist.Kumaraswamy, 10.0, np.array([2.0, 3.0])),
+    T(dist.Kumaraswamy, np.array([1.7]), np.array([[2.0], [3.0]])),
```
Comment (OlaRonning) [inline test/test_distributions.py:289]: Could you add an example with `0. < conc0 or conc1 < 1.`?

**Comment by OlaRonning** [comment]
@fehiepsi do you have an intuition about when to prefer `Kumaraswamy` over `beta`?

**Comment by fehiepsi** [comment]
Because Kumaraswamy can be reparameterized, it might be useful in SVI. Personally I found that it is fine to use Beta directly.

File: test/test_distributions.py (hunk 3 above)
Comment (fehiepsi) [reply test/test_distributions.py:289]: I just added the test.

**Review by fehiepsi — COMMENTED** [review]
Thanks for reviewing, @OlaRonning!

**Review by OlaRonning — APPROVED** [review]
LGTM. Thanks @fehiepsi

**Pull request merged**